cdap-app-fabric/src/test/java/io/cdap/cdap/metadata/ScheduleFetcherTest.java

```diff
@@ -27,27 +27,56 @@
 import io.cdap.cdap.proto.id.ProgramId;
 import io.cdap.cdap.proto.id.ScheduleId;
 import org.junit.Assert;
-import org.junit.BeforeClass;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
+import java.util.Arrays;
+import java.util.Collection;
 import java.util.List;
 
 /**
- * Tests for {@link RemoteScheduleFetcher}
+ * Tests for {@link RemoteScheduleFetcher} and {@link LocalScheduleFetcher}
  */
-public class RemoteScheduleFetcherTest extends AppFabricTestBase {
-  private static ScheduleFetcher fetcher = null;
+@RunWith(Parameterized.class)
+public class ScheduleFetcherTest extends AppFabricTestBase {
+  public enum ScheduleFetcherType {
+    LOCAL,
+    REMOTE,
+  };
 
-  @BeforeClass
-  public static void setUp() {
-    fetcher = getInjector().getInstance(RemoteScheduleFetcher.class);
+  private ScheduleFetcherType fetcherType = null;
```

Comment:
this can be private final, you don't need to assign null to it, since the constructor will assign value to it

Reply:
Done.